src/app/initTchap.ts

```diff
@@ -0,0 +1,89 @@
+import defaultDispatcher from "matrix-react-sdk/src/dispatcher/dispatcher";
+import { ActionPayload } from "matrix-react-sdk/src/dispatcher/payloads";
+import PlatformPeg from "matrix-react-sdk/src/PlatformPeg";
+
+import TchapVersionManagement from "../util/TchapVersionManagement";
+import TchapUserSettings from "../util/TchapUserSettings";
+import TchapUIFeature from "../util/TchapUIFeature";
+
+/**
+ * Determine weither the app needs a clearCacheAndReload after loading. We do it when upgrading from v2 to v4, to avoid weird keys bugs.
+ * @returns Promise(true) if a refresh is needed, Promise(false) in other cases
+ */
+export async function needsRefreshForVersion4(): Promise<boolean> {
+    //check if the feature is activated
+    if (!TchapUIFeature.activateClearCacheAndReloadAtVersion4) {
+        return Promise.resolve(false);
```

Comment:
You can just `return false` here because it is an `async` function. Reduces visual clutter.

Reply:
done, thanks !